homeassistant/components/homee/config_flow.py

```diff
@@ -0,0 +1,106 @@
+"""Config flow for homee integration."""
+
+import logging
+from typing import Any
+
+from pyHomee import (
+    Homee,
+    HomeeAuthFailedException as HomeeAuthenticationFailedException,
+    HomeeConnectionFailedException,
+)
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import AbortFlow
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+AUTH_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class HomeeConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for homee."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.homee: Homee = None
+        self.all_devices: bool = True
+        self.debug_data: bool = False
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial user step."""
+
+        errors = {}
+        if user_input is not None:
+            self.homee = Homee(
+                user_input[CONF_HOST],
+                user_input[CONF_USERNAME],
+                user_input[CONF_PASSWORD],
+            )
+
+            try:
+                await self.homee.get_access_token()
+                _LOGGER.info("Got access token for homee")
+            except HomeeConnectionFailedException:
+                errors["base"] = "cannot_connect"
+            except HomeeAuthenticationFailedException:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                self.hass.loop.create_task(self.homee.run())
+                _LOGGER.debug("Homee task created")
+                await self.homee.wait_until_connected()
+                _LOGGER.info("Homee connected")
+                self.homee.disconnect()
+                _LOGGER.debug("Homee disconnecting")
+                await self.homee.wait_until_disconnected()
+                _LOGGER.info("Homee config successfully tested")
+
+                await self.async_set_unique_id(self.homee.settings.uid)
+                try:
+                    self._abort_if_unique_id_configured()
+                except AbortFlow:
+                    return self.async_abort(reason="already_configured")
+                else:
+                    _LOGGER.info(
+                        "Created new homee entry with ID %s", self.homee.settings.uid
+                    )
+
+                    return self.async_create_entry(
+                        title=f"{self.homee.settings.uid} ({self.homee.host})",
```

Comment:
Why don't we use the homee name as title?

Reply:
Good question - was like this forever and I never questioned it. done.